src/adapters/MeetingsJSONAdapter.js

```diff
@@ -854,4 +888,70 @@ export default class MeetingsJSONAdapter extends MeetingsAdapter {
 
     return concat(initialState$, settingsEvent$);
   }
+
+  /**
+   * Returns an observable that emits the display data of the camera switcher control.
+   *
+   * @param {string} ID  Meeting ID
+   * @returns {Observable.<MeetingControlDisplay>} Observable that emits control display data
+   * @private
+   */
+  cameraSwitcherControl(ID) {
+    const meeting = this.datasource[ID];
+    const availableCameras$ = defer(() => this.getAvailableDevices('videoinput'));
+
+    const initialControl$ = new Observable((observer) => {
+      if (meeting) {
+        observer.next({
+          ID: SWITCH_CAMERA_CONTROL,
+          tooltip: 'Video Devices',
+          options: null,
+          selected: null,
+        });
+        observer.complete();
+      } else {
+        observer.error(new Error(`Could not find meeting with ID "${ID}" to add camera switcher control`));
+      }
+    });
+
+    const controlWithOptions$ = initialControl$.pipe(
+      concatMap((control) => availableCameras$.pipe(
+        map((availableCameras) => ({
+          ...control,
+          options: availableCameras && availableCameras.map((camera) => ({
```

Comment:
null means "loading", so in case there is an error from getAvailableCameras we should put an empty list:
```js
options: (availableCameras || []).map((camper) => ...
```